src/tools/optimization-options.h

```diff
@@ -202,12 +202,28 @@ struct OptimizationOptions : public ToolOptions {
            });
     // add passes in registry
     for (const auto& p : PassRegistry::get()->getRegisteredNames()) {
-      (*this).add(
-        std::string("--") + p,
-        "",
-        PassRegistry::get()->getPassDescription(p),
-        Options::Arguments::Zero,
-        [this, p](Options*, const std::string&) { passes.push_back(p); });
+      (*this).add(std::string("--") + p,
+                  "",
+                  PassRegistry::get()->getPassDescription(p),
+                  // Allow an optional parameter to a pass. If provided, it is
+                  // the same as if using --pass-arg, that is,
+                  //
+                  //   --foo=ARG
+                  //
+                  // is the same as
+                  //
+                  //   --foo --pass-arg=foo@ARG
+                  Options::Arguments::Optional,
+                  [this, p](Options*, const std::string& arg) {
+                    if (!arg.empty()) {
+                      if (passOptions.arguments.count(p)) {
+                        Fatal()
+                          << "Cannot pass multiple pass arguments to " << p;
+                      }
+                      passOptions.arguments[p] = arg;
```

Comment:
I've been a big fan of `insert` to reduce map lookups recently, but if you think the other way is more readable, that's probably fine, too.

```suggestion
                      if (!passOptions.arguments.insert({p, arg}).second) {
                        Fatal()
                          << "Cannot pass multiple pass arguments to " << p;
                      }
```

Reply:
I'd like `insert` more if the `.second` were `.inserted`... as things stand it requires mental work when I read such a line. Just me?

Reply:
Yes, totally agree that it's a terrible API. My hope is that if we use it more, we won't need to think about it as much when we see it. I don't think it's important in this particular case, though.

Reply:
Fair point. Maybe we should use it more. Or maybe we could wrap around it with something clearer. Let's leave that separately though.